core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/ResultColumnMixin.kt

```diff
@@ -49,4 +51,13 @@ internal abstract class ResultColumnMixin(
   }
 
   override fun queryExposed() = queryExposed.forFile(containingFile)
+
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    super.annotate(annotationHolder)
+
+    val bindExpr = expr as? SqlBindExpr
+    if (bindExpr != null) {
+      annotationHolder.createErrorAnnotation(bindExpr, "Cannot bind the name of a column in a SELECT result-column")
+    }
```

Comment:
I did it like this because smartcast can't be used on expr. I can do it as a let if you think that would look better.

Reply:
i think this is good 👍